cli/src/commands/help.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::Write;
+
+use tracing::instrument;
+
+use crate::cli_util::CommandHelper;
+use crate::command_error::CommandError;
+use crate::ui::Ui;
+
+/// Print this message or the help of the given subcommand(s)
+#[derive(clap::Args, Clone, Debug)]
+pub(crate) struct HelpArgs {
+    /// The command to show help
+    command: Option<String>,
+}
+
+#[instrument(skip_all)]
+pub(crate) fn cmd_help(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &HelpArgs,
+) -> Result<(), CommandError> {
+    let help_str = if let Some(command_to_show_help) = &args.command {
+        command
+            .app()
+            .find_subcommand(command_to_show_help.clone())
+            .ok_or(command.app().clone().error(
+                clap::error::ErrorKind::InvalidValue,
+                format!("no subcomand with name {}", command_to_show_help),
+            ))?
+            .clone()
+            .render_help()
+    } else {
+        command.app().clone().render_help()
+    };
+
+    let help_str = if ui.color() {
+        help_str.ansi().to_string()
+    } else {
+        help_str.to_string()
+    };
```

Comment:
It's probably better to call `handle_clap_error()` here. That reduces some duplication and means that pager support still works (when `--no-pager` is not passed).

Reply:
Maybe even better, would be to just return with a `CommandError::Cli` with the `DisplayHelp` inside it?

Reply:
Yes, that does seem better. Good point :)